src/modules/RA/DataLayer/components/DataLayerForm.js

```diff
@@ -13,12 +13,26 @@ import MinisheetTab from './tabs/MinisheetTab';
 import FilterTab from './tabs/FilterTab';
 import TableTab from './tabs/TableTab';
 
-const DataLayerForm = ({ ...props }) => {
+const initialErrorState = {
+  definition: false,
+  style: false,
+  legend: false,
+  popup: false,
+  minisheet: false,
+  filter: false,
+  table: false,
+};
+
+const inErrorReducer = (state, { type, payload }) => {
+  if (state[type] === Boolean(payload)) {
+    return state;
+  }
+  return { ...state, [type]: Boolean(payload) };
```

Comment:
Do you really need to spread `state` when setting the `[type]`?

Reply:
I think you have to, or you will lose the `errorState` for the other type (i.e other tabs)

Reply:
Okay, it looks good to me 👍